syntax = "proto3";

package com.cart.grpc;

import "google/protobuf/empty.proto";

service CartService {
  // Adiciona um item ao carrinho
  rpc AddToCart (CartItemRequest) returns (google.protobuf.Empty);

  // Remove um item do carrinho
  rpc RemoveFromCart (CartItemRequest) returns (google.protobuf.Empty);

  // Retorna todos os itens do carrinho de um usuário
  rpc GetCart (UserQuery) returns (Cart);

  // Limpa todos os itens do carrinho
  rpc ClearCart (UserQuery) returns (google.protobuf.Empty);
}

// Consulta baseada no ID do usuário
message UserQuery {
  string user_id = 1;
}

// Requisição para adicionar ou remover item do carrinho
message CartItemRequest {
  string user_id = 1;
  string product_name = 2;
  int32 quantity = 3;
}

// Item do carrinho (inclui nome, preço e quantidade)
message CartItem {
  string product_name = 1;
  float price = 2;
  int32 quantity = 3;
}

// Representa um carrinho inteiro
message Cart {
  string user_id = 1;
  repeated CartItem items = 2;
}
